From: Bram Moolenaar <Bram@vim.org>
Date: Sun, 9 Jul 2017 11:07:16 +0200
Subject: patch 8.0.0703: illegal memory access with empty :doau command

Problem:    Illegal memory access with empty :doau command.
Solution:   Check the event for being out of range. (James McCoy)
diff --git a/src/fileio.c b/src/fileio.c
index ed8d45b..eb34ec2 100644
--- a/src/fileio.c
+++ b/src/fileio.c
@@ -8685,7 +8685,7 @@ do_doautocmd(
     /*
      * Loop over the events.
      */
-    while (*arg && !vim_iswhite(*arg))
+    while (*arg && !ends_excmd(*arg) && !vim_iswhite(*arg))
 	if (apply_autocmds_group(event_name2nr(arg, &arg),
 				      fname, NULL, TRUE, group, curbuf, NULL))
 	    nothing_done = FALSE;
@@ -9192,7 +9192,8 @@ apply_autocmds_group(
      * Quickly return if there are no autocommands for this event or
      * autocommands are blocked.
      */
-    if (first_autopat[(int)event] == NULL || autocmd_blocked > 0)
+    if (event == NUM_EVENTS || first_autopat[(int)event] == NULL
+	    || autocmd_blocked > 0)
 	goto BYPASS_AU;
 
     /*
@@ -9265,7 +9266,7 @@ apply_autocmds_group(
     {
 	if (event == EVENT_COLORSCHEME || event == EVENT_OPTIONSET)
 	    autocmd_fname = NULL;
-	else if (fname != NULL && *fname != NUL)
+	else if (fname != NULL && !ends_excmd(*fname))
 	    autocmd_fname = fname;
 	else if (buf != NULL)
 	    autocmd_fname = buf->b_ffname;
